exchanges/bybit/bybit.go

```diff
@@ -0,0 +1,746 @@
+		amount, err := strconv.ParseFloat(ob[x][1], 64)
+		if err != nil {
+			return nil, err
+		}
+		price, err = strconv.ParseFloat(ob[x][0], 64)
+		if err != nil {
+			return nil, err
+		}
+		o = append(o, OrderbookItem{
+			Price:  price,
+			Amount: amount,
+		})
+	}
+	return o, nil
+}
+
+type orderResponse struct {
+	Data struct {
+		Asks [][2]string `json:"asks"`
+		Bids [][2]string `json:"bids"`
+		Time int64       `json:"time"`
+	} `json:"result"`
+}
+
+// GetOrderbook gets orderbook for a given market with a given depth (default depth 100)
+func (by *Bybit) GetOrderBook(ctx context.Context, symbol string, depth int64) (Orderbook, error) {
+	var order orderResponse
+	strDepth := "100" // default depth
+	if depth > 0 && depth < 100 {
+		strDepth = strconv.FormatInt(depth, 10)
+	}
+
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	params.Set("limit", strDepth)
+	path := common.EncodeURLValues(bybitOrderBook, params)
+	err := by.SendHTTPRequest(ctx, exchange.RestSpot, path, publicSpotRate, &order)
+	if err != nil {
+		return Orderbook{}, err
+	}
+
+	var s Orderbook
+	s.Bids, err = processOB(order.Data.Bids)
+	if err != nil {
+		return s, err
+	}
+	s.Asks, err = processOB(order.Data.Asks)
+	if err != nil {
+		return s, err
+	}
+	s.Time = time.UnixMilli(order.Data.Time)
+	return s, nil
+}
+
+// GetMergedOrderBook gets orderbook for a given market with a given depth (default depth 100)
+func (by *Bybit) GetMergedOrderBook(ctx context.Context, symbol string, scale, depth int64) (Orderbook, error) {
+	var order orderResponse
+	params := url.Values{}
+	if scale > 0 {
+		params.Set("scale", strconv.FormatInt(scale, 10))
+	}
+
+	strDepth := "100" // default depth
+	if depth > 0 && depth < 100 {
+		strDepth = strconv.FormatInt(depth, 10)
+	}
+
+	params.Set("symbol", symbol)
+	params.Set("limit", strDepth)
+	path := common.EncodeURLValues(bybitMergedOrderBook, params)
+	err := by.SendHTTPRequest(ctx, exchange.RestSpot, path, publicSpotRate, &order)
+	if err != nil {
+		return Orderbook{}, err
+	}
+
+	var s Orderbook
+	s.Bids, err = processOB(order.Data.Bids)
+	if err != nil {
+		return s, err
+	}
+	s.Asks, err = processOB(order.Data.Asks)
+	if err != nil {
+		return s, err
+	}
+	s.Time = time.UnixMilli(order.Data.Time)
+	return s, nil
+}
+
+// GetTrades gets recent trades from the exchange
+func (by *Bybit) GetTrades(ctx context.Context, symbol string, limit int64) ([]TradeItem, error) {
+	resp := struct {
+		Data []struct {
+			Price        float64 `json:"price,string"`
+			Time         int64   `json:"time"`
+			Quantity     float64 `json:"qty,string"`
+			IsBuyerMaker bool    `json:"isBuyerMaker"`
+		} `json:"result"`
+	}{}
+
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	params.Set("limit", strconv.FormatInt(limit, 10))
+	path := common.EncodeURLValues(bybitRecentTrades, params)
+	err := by.SendHTTPRequest(ctx, exchange.RestSpot, path, publicSpotRate, &resp)
+	if err != nil {
+		return nil, err
+	}
+
+	var trades []TradeItem
```

Comment:
As above we can pre-allocate most of these slices using make with the len of resp.Data. This reduces the need for re-allocation on append when we know what the max data could be, could you please go across the PR and make those changes.